### PR TITLE
Bump mariadb-java-client to 3.0.7

### PR DESCRIPTION
**Pull request by Wooza**
Bumped mariadb-java-client version to 3.0.7 to see which tests may break (see #2782)

### CODE REVIEW

**Comment by Wooza** [comment]
Given that all tests passed I would assume that the relevant code path was not taken by any of the tests? I'm not sure about your CI setup but there seems to be no coverage data to verify this?

**Comment by rbygrave** [comment]
> relevant code path was not taken by any of the tests?

It's more that we have different github actions for the different databases and we don't run those database specific actions on each PR.  So the PR's are run against H2 database.

Normally I can choose the MariaDB action and run that against a branch ... but it isn't seeing this PR branch. Hmmm.

**Comment by rbygrave** [comment]
Yeah, I can't run the action against the forked branch.  I'll look to merge this in and then run using the MariaDB action and that should fail.

Background:
From memory ebean is using that weird setting to get mysql jdbc driver to "stream" results rather than "pull all results into local buffer" and ebean does this for findEach, findStream, findIterate queries. There is also another "bit of a hack/limitation" we hit with mysql around firing secondary queries while streaming results so perhaps the mariadb driver also fixed that issue.

File: ebean-core/src/main/java/io/ebeaninternal/server/query/CQuery.java [hunk 1]
```diff
@@ -314,7 +314,7 @@ ResultSet prepareResultSet(boolean forwardOnlyHint) throws SQLException {
       if (forwardOnlyHint) {
         // Use forward only hints for large resultSet processing (Issue 56, MySql specific)
         pstmt = conn.prepareStatement(sql, ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_READ_ONLY);
-        pstmt.setFetchSize(Integer.MIN_VALUE);
+        pstmt.setFetchSize(1);
```
Comment (rbygrave) [inline ebean-core/src/main/java/io/ebeaninternal/server/query/CQuery.java:317]: We can't do this part of course because that breaks MySql.

**Pull request merged**